lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -295,8 +295,9 @@
     - ' - C(dns_servers) (list): List of DNS servers to configure.'
     - ' - C(dns_suffix) (list): List of domain suffixes, also known as DNS search path (default: C(domain) parameter).'
     - ' - C(domain) (string): DNS domain name to use.'
-    - ' - C(hostname) (string): Computer hostname (default: shorted C(name) parameter). Allowed characters are alphanumeric (uppercase and lowercase)
-          and minus, rest of the characters are dropped as per RFC 952.'
+    - ' - C(hostname) (string): Computer hostname (default: C(name) parameter). Allowed characters are alphanumeric (uppercase and lowercase)
+          and minus (no dots allowed, VMware guest customization for linux does not support receiving a FQDN name). Use "force=True" for old
```

Comment:
Making the new behavior the default is not backwards compatible. The new behavior should be opt-in rather than opt-out to avoid causing problems with existing playbooks during an upgrade.

Reply:
An important general rule indeed, but IMHO this should be an exception. 

Because I believe the current behaviour is misleading. Not sure if it's the best word to describe, but It made me waste time figuring out why the domain name was not being configured in the OS hostname as I had instructed the module to do. Instead it could just have thrown an error telling me it wasn't supported. Can happen to many others, probably have already.

I also believe that the likeliness of someone out there relying on this behaviour is not that big. It could surely be masking an configuration error that will be revealed with this, but should be an easy fix, and people should validate/test new ansible versions before adopting then anyway.

Could you give use some advice on that @Akasurde ?

If we make the new behaviour opt-in, then it would need to be something more than just trigger the new behaviour  with --force, because that would be also confusing. Any suggestions in that case?

Reply:
Be sure to include a changlog entry explaining the backwards compatibility issue.